lib/url.js

```diff
@@ -587,11 +594,35 @@ function urlFormat(urlObject, options) {
   } else if (typeof urlObject !== 'object' || urlObject === null) {
     throw new ERR_INVALID_ARG_TYPE('urlObject',
                                    ['Object', 'string'], urlObject);
+  } else if (urlObject instanceof URL) {
+    let fragment = true;
+    let unicode = false;
+    let search = true;
+    let auth = true;
+
+    if (options) {
+      validateObject(options, 'options');
+
+      if (options.fragment != null) {
+        fragment = Boolean(options.fragment);
+      }
+
+      if (options.unicode != null) {
+        unicode = Boolean(options.unicode);
+      }
+
+      if (options.search != null) {
+        search = Boolean(options.search);
+      }
+
+      if (options.auth != null) {
+        auth = Boolean(options.auth);
+      }
+    }
+
+    return formatUrl(urlObject.href, fragment, unicode, search, auth);
   } else if (!(urlObject instanceof Url)) {
```

Comment:
```suggestion
  } else {
```

I think

Reply:
I can directly call `Url.prototype.format.call(urlObject);` and remove `urlObject.format()` from the last line, if this is what you recommended?

Reply:
Well ... it seems like the last line here is unreachable now, so, yeah, if it doesn't make any difference, feel free to remove it